run/src/master/master.cfg

```diff
@@ -0,0 +1,295 @@
+
+# 'protocols' contains information about protocols which master will use for
+# communicating with workers. You must define at least 'port' option that workers
+# could connect to your master with this protocol.
+# 'port' must match the value configured into the workers (with their
+# --master option)
+c['protocols'] = {'pb': {'port': 9989}}
+
+####### CHANGESOURCES
+
+# the 'change_source' setting tells the buildmaster how it should find out
+# about source code changes.  Here we point to the buildbot version of a python hello-world project.
+
+c['change_source'] = [
+    changes.GitPoller('git://github.com/w3c/web-platform-tests.git',
+                      workdir='gitpoller-workdir',
+                      branch='master',
+                      pollAtLaunch=True,
+                      pollinterval=300)
+]
+
+@util.renderer
+def render_chunked_builder(properties):
+    if properties.getProperty('use_sauce_labs'):
+        return ['Remote Chunked Runner']
+
+    return ['Local Chunked Runner']
+
+c['schedulers'] = [
+  schedulers.Triggerable(name='chunked',
+                         builderNames=render_chunked_builder),
+  schedulers.Triggerable(name='upload',
+                         builderNames=['Uploader']),
+  schedulers.Nightly(name='Bi-daily (slow builds)',
+                     builderNames=['Chunk Initiator'],
+                     onlyIfChanged=False,
+                     properties={'build_speed': 'slow'},
+                     dayOfMonth=range(0, 31, 2),
+                     hour=0),
+  schedulers.Nightly(name='Semi-daily (fast builds)',
+                     builderNames=['Chunk Initiator'],
+                     onlyIfChanged=False,
+                     properties={'build_speed': 'fast'},
+                     hour=[0, 6, 12, 18])
+]
+
+chunked_steps = []
+
+def filter_build(speed, step):
+    return speed == step.build.properties.getProperty('build_speed')
+
+for spec in platform_manifest.values():
+    if spec.get('currently_run') != True:
+        continue
+
+    speed = 'slow' if spec.get('sauce') else 'fast'
+
+    chunked_steps.append(WPTChunkedStep(schedulerNames=['chunked'],
+                                        platform=spec,
+                                        doStepIf=partial(filter_build, speed),
+                                        total_chunks=100))
+
+trigger_factory = util.BuildFactory(
+    [
+        # Fetch the latest version of `master` prior to triggering chunked
+        # runs.  This ensures that the all chunks use the same revision of WPT
+        # (by setting the Buildbot property named `revision`), including when
+        # failed builds are manually re-tried via the web interface.
+        steps.Git(repourl='git://github.com/w3c/web-platform-tests'),
+        steps.SetPropertyFromCommand(name='Retrieve commit date of latest revision',
+                                     command=['git', 'log', '-1', '--format=%cd', '--date=iso-strict'],
```

Comment:
Please add a comment explaining why this code is capturing the date of the commit now, instead of at the end.

Reply:
Sure